rcode-reader/src/main/java/eu/rssw/pct/RCodeInfo.java

```diff
@@ -0,0 +1,723 @@
+    int bytesRead = input.read(segment);
+    if (bytesRead != debugSegmentSize) {
+      throw new InvalidRCodeException("Not enough bytes in debug segment block");
+    }
+    if (out != null) {
+      out.printf("%n*******%nDEBUG%n*******%n");
+      printByteBuffer(out, segment);
+    }
+
+  }
+
+  public RCodeUnit getUnit() {
+    return unit;
+  }
+
+  public static void printByteBuffer(PrintStream writer, byte[] block) {
+    StringBuilder sb = new StringBuilder();
+    int pos = 0;
+    while (pos < block.length) {
+      if ((pos % 16) == 0) {
+        writer.print(String.format("%010X | ", pos));
+      }
+      writer.print(String.format("%02X ", block[pos]));
+      if (Character.isISOControl((char) block[pos])) {
+        sb.append('.');
+      } else {
+        sb.append((char) block[pos]);
+      }
+      if ((pos > 0) && (((pos + 1) % 16) == 0)) {
+        writer.println(" | " + sb.toString());
+        sb = new StringBuilder();
+      }
+      pos++;
+    }
+    if ((pos % 16) != 0) {
+      writer.print(Strings.repeat("   ", 16 - (pos % 16)));
+      writer.println(" | " + sb.toString());
+    }
+  }
+
+  /**
+   * Returns r-code compiler version
+   * 
+   * @return Version
+   */
+  public long getVersion() {
+    return version;
+  }
+
+  /**
+   * Returns r-code timestamp (in milliseconds)
+   * 
+   * @return Timestamp
+   */
+  public long getTimeStamp() {
+    return timeStamp;
+  }
+
+  public boolean is64bits() {
+    return sixtyFourBits;
+  }
+
+  public boolean isClass() {
+    return isClass;
+  }
+
+  public String getTypeName() {
+    return typeName;
+  }
+
+  public String getParentTypeName() {
+    return parentTypeName;
+  }
+
+  public String getAssemblyName() {
+    return assemblyName;
+  }
+
+  public List<String> getInterfaces() {
+    return interfaces;
+  }
+
+  public static String readNullTerminatedString(byte[] array, int offset) {
+    return readNullTerminatedString(array, offset, Charset.defaultCharset());
+  }
+
+  public static String readNullTerminatedString(byte[] array, int offset, Charset charset) {
+    int zz = 0;
+    while ((zz + offset < array.length) && (array[zz + offset] != 0)) {
+      zz++;
+    }
+
+    return charset.decode(ByteBuffer.wrap(array, offset, zz)).toString();
+  }
+
+  private static int readAsciiEncodedNumber(byte[] array, int pos, int length) throws InvalidRCodeException {
+    try {
+      return Integer.valueOf(new String(Arrays.copyOfRange(array, pos, pos + length)), 16);
+    } catch (NumberFormatException caught) {
+      throw new InvalidRCodeException(caught);
+    }
+  }
+
+  public static class InvalidRCodeException extends Exception {
+    private static final long serialVersionUID = 1L;
+
+    public InvalidRCodeException(String s) {
+      super(s);
+    }
+
+    public InvalidRCodeException(Throwable caught) {
+      super(caught);
+    }
+  }
+
+  private static Function parseProcSignature(String str) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused private "parseProcSignature" method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AUnusedPrivateMethod)